core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -543,11 +579,30 @@ public void unmount(AlluxioURI path, UnmountPOptions options)
     });
   }
 
+  private void updateFileAccessTime(AlluxioURI path) throws IOException, AlluxioException {
```

Comment:
This logic should not be in `BaseFileSystem`?

Reply:
it's for the new `openFile(URIStatus)` API

Reply:
This functionality doesn't feel like it belongs in the base client because the base file system shouldn't support the `openFile(URIStatus,...)` API. Instead, could we only add this new API to `CachingFileSystem`?

Reply:
this API is added to the FileSystem interface to support the following calling pattern:
```
status = getStatus;
// do something with status
stream = openFile(status);
```
This pattern is seen in `AlluxioFuseFileSystem`, and it's orthogonal to caching.

Reply:
I'm against introducing this API to the file system
1. We already have openFile which handles getting the file status, this sounds like an optimization for when the file status is already available (ie. cached)
2. It adds complexity to the base file system class by introducing async operations
3. This could easily be included in another file system type (that extends BaseFileSystem) and provide the same benefits to users

Reply:
Refactored exception handling of `openFile(AlluxioURI)`, so that `AlluxioFuseFileSystem` gets rid of `openFile(Status)`.